File: turms-client-swift/Sources/TurmsClient/Model/SessionCloseStatus.swift
import Foundation

public enum SessionCloseStatus: Int {
    case illegalRequest = 100
    case heartbeatTimeout = 110
    case loginTimeout
    case switchProtocol

    case serverError = 200
    case serverClosed
    case serverUnavailable

    case connectionClosed = 300

    case unknownError = 400

    case disconnectedByClient = 500
    case disconnectedByOtherDevice

    case disconnectedByAdmin = 600

    case userIsDeletedOrInactivated = 700
    case userIsBlocked
}

File: turms-client-dart/lib/src/model/session_close_status.dart
class SessionCloseStatus {
  static const illegalRequest = 100;
  static const heartbeatTimeout = 110;
  static const loginTimeout = 111;
  static const switchProtocol = 112;

  static const serverError = 200;
  static const serverClosed = 201;
  static const serverUnavailable = 202;

  static const connectionClosed = 300;

  static const unknownError = 400;

  static const disconnectedByClient = 500;
  static const disconnectedByOtherDevice = 501;

  static const disconnectedByAdmin = 600;

  static const userIsDeletedOrInactivated = 700;
  static const userIsBlocked = 701;
}
